recipes/libssh/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+import os
+
+
+required_conan_version = ">=1.54.0"
+
+
+class LibSSHRecipe(ConanFile):
+    name = "libssh"
+    license = "LGPL-2.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.libssh.org/"
+    description = "multiplatform C library implementing the SSHv2 protocol on client and server side"
+    topics = ("ssh", "shell", "ssh2", "connection")
+    package_type = "library"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "crypto_backend": ["openssl", "gcrypt"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "crypto_backend": "openssl",
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/[>=1.2.11 <2]")
+        if self.options.crypto_backend =="openssl":
+            self.requires("openssl/[>=1.1 <4]")
+        elif self.options.crypto_backend == "gcrypt":
+            self.requires("libgcrypt/1.8.4")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CLIENT_TESTING"] = False
+        tc.variables["SERVER_TESTING"] = False
+        tc.variables["WITH_EXAMPLES"] = False
+        tc.variables["WITH_GCRYPT"] = self.options.crypto_backend == "gcrypt"
+        tc.variables["WITH_GSSAPI"] = False
+        tc.variables["WITH_MBEDTLS"] = False
+        tc.variables["WITH_NACL"] = False
+        tc.variables["WITH_SYMBOL_VERSIONING"] = False
+        tc.variables["WITH_ZLIB"] = self.options.with_zlib
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.cache_variables["CMAKE_TRY_COMPILE_CONFIGURATION"] = str(self.settings.build_type)
```

Comment:
Nice! This indeed seems to be the correct workaround for this CMake/Conan issue:  https://github.com/conan-io/conan/issues/12180